### PR TITLE
Interim definition

### LINKED ISSUE #226: Clarify the definition of status `interim`
During the review of #205 there were some comments regarding the definitions for the `/document/status` value `interim`. The current definition is provided to ease the discussions:

> The value `final` indicates, that the issuing party asserts the content is unlikely to change. “Final” status is an indication only, and does not preclude updates.
>
> The value `interim` indicates, that the issuing party asserts the content is unlikely to change.

---

@tschmidtb51 comment in https://github.com/oasis-tcs/csaf/pull/205#discussion_r612818475:

> @sthagen: This looks like the same definition as `final`. I would suggest the following:
> ```suggestion
> The value `interim` indicates, that the issuing party asserts the content of the current version is unlikely to change. However, further updates of this advisory are already planned.
>```

---

@mprpic replied in https://github.com/oasis-tcs/csaf/pull/205#discussion_r612864616:

> "Content is unlikely to change" and "advisory is planned to be updated" kind of seem like opposite statements. I'm actually not sure what the purpose of the "interim" status is. Either the advisory is in draft and may change, or it's in its final form and is ready for consumption. As a consumer, I'm not sure I would care about a third state when even the "final" state does not mean the content won't ever change again (that's what revision history is for).

---

@sthagen repiled in https://github.com/oasis-tcs/csaf/pull/205#discussion_r612951334:

> I would think of amending an already present content with additional content or measures? But here better practitioners having to produce the incremental versions do speak up.

---

To me, there are 2 options:
1. `interim` is a state which is used for all advisories from the `/document/tracking/initial_release_date` until the issuing party states that no further updates will follow. Then the last advisory gets the status `final`. This would include a change of the definition of `final` to make clear that an advisory, once reached the state final should only be updated in exceptional cases.

   **Pros:** The customer could easily determine whether the issuer intends to update the advisory.
   **Cons:** This could lead to issuers leaving advisories in status `interim` forever :disappointed: 

2. `interim` is a state used in the generation of an advisory as described in conformance target "CSAF content management system" under workflows:

   >- "New Advisory": create a new advisory, request a review, provide review comments or approve it, resolve review comments; if the review approved it (draft->interim), the approval for publication can be requested; if granted (manual or time-based) the advisory is provided for publication (interim -> final)
   >- "Update Advisory": open an existing advisory, create new revision & change content (interim), request a review, provide review comments or approve it, resolve review comments; if the review approved it (draft->interim), the approval for publication can be requested; if granted (manual or time-based) the updated advisory is provided for publication (interim -> final)

   **Pros:** This allows a better coordination during the process of generating an advisory as this might involve the researcher, vendors and coordinating agencies (and sometimes also a user). Moreover, it provides the opportunity to pre-share advisories (e.g. with authorities) before they are final.
   **Cons:** At the moment, I can't think of any.

### PR DESCRIPTION
**Pull request by tschmidtb51**
- fix #226 
- change definition of "interim"
- add informational remark about pulling versions
- fix wrong usage of "interim" in CSAF content management system
- add rules to support document status "interim" in CSAF content management system

### CODE REVIEW

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 1]
```diff
@@ -1613,11 +1613,13 @@ The value MUST be one of the following:
     interim
 ```
 
-The value `draft` indicates, that this is a pre-release, intended for issuing party’s internal use only, or possibly used externally when the party is seeking feedback or indicating its intentions regarding a specific issue.
+The value `draft` indicates, that this is a pre-release, intended for issuing party's internal use only, or possibly used externally when the party is seeking feedback or indicating its intentions regarding a specific issue.
 
-The value `final` indicates, that the issuing party asserts the content is unlikely to change. “Final” status is an indication only, and does not preclude updates.
+The value `final` indicates, that the issuing party asserts the content is unlikely to change. “Final” status is an indication only, and does not preclude updates. This SHOULD be used if the issuing party no, slow or few changes.
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:1618]: There appears to be a typo that blocks me from understanding the intent of the sentence:
> [...]if the issuing party no, slow or few changes.

File: csaf_2.0/prose/csaf-v2-editor-draft.md [hunk 2]
```diff
@@ -1613,11 +1613,13 @@ The value MUST be one of the following:
...
 
-The value `interim` indicates, that the issuing party asserts the content is unlikely to change.
+The value `interim` indicates, that the issuing party asserts expect rapid updates. This SHOULD be used if the expected rate of release for this document is significant higher than for other documents. Once the rate slows down it MUST be changed to `final`. This may be done in a patch version.
```
Comment (sthagen) [inline csaf_2.0/prose/csaf-v2-editor-draft.md:1620]: Similar here (was `expects` meant to replace `asserts` ?):
> [...]the issuing party asserts expect rapid updates.

**Review by sthagen — APPROVED** [review]
When the two nits are fixed we should merge. LGTM.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 2 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md:1620]: Yes.

File: csaf_2.0/prose/csaf-v2-editor-draft.md (hunk 1 above)
Comment (tschmidtb51) [reply csaf_2.0/prose/csaf-v2-editor-draft.md]: I hope the inserted word makes it clear. :-)

**Review by sthagen — APPROVED** [review]
LGTM.

**Review by santosomar — APPROVED** [review]

**Pull request merged**